### ISSUE TITLE
Add example on invoking lambda functions

### ISSUE THREAD

**Issue by ghost** [BruceBuckland/SignIn-awsmhh#5]
I've used SignIn-awsmhh for my project to work using Cognito Federated Identities with Cognito User Pools, starting from a new project.

Now user is logged-in and I want to invoke lambda function and this lead to a crash:
`terminating with uncaught exception of type NSException`

Logs displayed the following useful informations before crash occurs:
```
AWSiOSSDK v2.4.11 [Debug] AWSInfo.m line:122 | -[AWSServiceInfo initWithInfoDictionary:checkRegion:] | Couldn't read the region configuration from Info.plist for the client. Please check your `Info.plist` if you are providing the SDK configuration values through `Info.plist`.
 *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'The service configuration is `nil`. You need to configure `Info.plist` or set `defaultServiceConfiguration` before using this method.'
```

And here is the code:

```
        let lambda = AWSLambda.defaultLambda()
        
        lambda.invoke(request, completionHandler: {
            (response: AWSLambdaInvocationResponse?, error: NSError?) in
            
            print("ERROR HERE: \(error)")
            let payload = response?.payload
            print("PAYLOAD HERE: \(payload)")
            
            })
```
AWS Mobile Hub console does not offer any settings or help to directly manage lambda. Nor informations on how to configure `Info.plist` as stated by logs.

So could you please add an example of lambda invocation using AWS Mobile Hub Helper?

**Comment by BruceBuckland** [comment]
@Domsware  

A service configuration is the basic element of control for asking Amazon Web services to do something .   It has the credentials provider associated with it .  And you can set the default  Service configuration to a particular value once and only once 

 I suspect you have not set the default service configuration somehow?

I looked at AWSIdentityManager and when you instantiate the singleton (call AWSIdentityManager.defaultIdentityManager ) the init for that singleton creates a service configuration for the identity manager and connects the identity manager as the authentication provider

I checked the cloudlogic code and it's init code creates a service configuration for the cloudlogic (lambda)...  connects the credentials provider and registers cloud logic with the service config... so if you use the cloud logic invoke in an app based on mobile hub helper it will do it for you.

You can call lambda directly, but the mobile hub has  a bunch of nice singletons that handle a lot of stuff for you... so I would suggest using that.

That is what the example does.

**Comment by BruceBuckland** [comment]
MySampleApp has a lambda example ( under Demo in MySampleApp-Sources/Demo/CloudLogic)

The documentation is thin, but not wrong.

https://docs.aws.amazon.com/awsmobilehubhelper/apireference/latest/index.html

This helps too.

http://docs.aws.amazon.com/mobile-hub/latest/developerguide/what-is-mobile-hub.html

**Comment by ghost** [comment]
> I suspect you have not set the default service configuration somehow?

As I use Mobile Hub Helper, this is already done.

I checked your MySampleApp demo for example. And integrate it in my code:

```
let cloudLogic = AWSCloudLogic.defaultCloudLogic()
cloudLogic.invokeFunction(...

```

I then update `Info.plist` by adding settings related to CloudLogic:

```
<key>CloudLogic</key>
        <dict>
            <key>Default</key>
            <dict>
                <key>Region</key>
                <string>eu-west-1</string>
            </dict>
        </dict>
```

I've checked you sample code and there nothing more about configuration.

It still not work... I downloaded sample app for AWS Mobile Hub to understand how things work. This is not the same way as in you sample code. 👎 

`AWSMobileClient.swift` has been updated to v0.5 as is the one from you sample code is v0.4
`// Source code generated from template: aws-my-sample-app-ios-swift v0.5`

**Comment by BruceBuckland** [comment]
@Domsware Oui! 
They have recently changed the Mobile Hub (but have not updated the aws-mobilehub-helper-ios github archive yet.  They say they will "soon").  The primary difference I have noted is that they have included a User Pools signin provider and a new signin provider factory class.  

Mais I compared the downloaded code from V0.4 and V0.5 and other than version number and white space the CloudLogic view controller is unchanged.

You said: "**As I use Mobile Hub Helper, this is already done**."
but this is not true in the code you have shown.  It is not enough to use only the AWSIdentityManager you must use the singletons from the other classes as well (**because it is in the constructors of those singletons that the service configuration is set up**)

I am not experienced in Lambda (at all!).  So take this with a grain of salt:

I note that in your code:

```
let lambda = AWSLambda.defaultLambda()

        lambda.invoke(request, completionHandler: {
            (response: AWSLambdaInvocationResponse?, error: NSError?) in
```

you establish the AWSLambda class... However with mobile hub helper that class is wrapped, and you should establish the AWSCloudLogic class.  

```
        AWSCloudLogic.defaultCloudLogic().invokeFunction(functionName,
            withParameters: parameters, completionBlock: {(result: AnyObject?, error: NSError?) -> Void in
```

Make sense?

Here is a pictorial overview of the mobile hub singletons
[mobile hub diagram.pdf](https://github.com/BruceBuckland/SignIn-awsmhh/files/597878/mobile.hub.diagram.pdf)

**Comment by ghost** [comment]
OK, everything is fine now using CloudLogic.

**Comment by BruceBuckland** [comment]
Excellent!

**Comment by BruceBuckland** [comment]
@Domsware It has been a pleasure working with you. Please accept the answer i gave over in stackexchange [ Link to your SE question ](url) ( I am trying to get rid of ads there). Also I checked out your site, and see that you have published several apps.  As I finish the app I am working on (it is for EMT's) and begin to try to get it on the app store, I was wondering if I might be able to contact you for advice.  Would that be ok?

**Comment by ghost** [comment]
You really helped me a lot on this project! Of course you can contact me for advice: it will be an honour to help you!
Let's contact me using [this](http://www.domsware.com/en/contact) contact form.
